### ISSUE TITLE
Exception not working if code called from IRQ

### ISSUE THREAD

**Issue by valentinbreiz** [CosmosOS/Cosmos#1769]
Exception are not handled in try catch if an exception is thrown from a code called by IRQ

**Comment by JuliusCaesar12** [comment]
Yes I have noticed this. I was running several PITTimers as makeshift threads and one caused a NullReferenceException in a try-catch, but instead of handling the error, the CPU halted due to an Invalid OpCode.

**Comment by DevBadger10** [comment]
I don't know what this means, but it sounds important.

**Comment by Kiirx** [comment]
It is

**Comment by valentinbreiz** [comment]
Cosmos Gen3 has replaced the IL2CPU-based Gen2 codebase, and `gen3` is now the default branch of this repository. This issue was filed against Gen2, which is no longer developed, so it is being closed as part of the Gen3 cutover (#3088).

If you still hit this on Gen3, please open a fresh issue. The compiler, the build pipeline and the debugging story are all different now, so a new report will get a far more useful answer than reviving this one. Start here: https://cosmosos.github.io/

Gen2 remains on the `gen2` branch, and this issue stays readable.
